lua/astrocommunity/pack/haskell/init.lua

```diff
@@ -12,14 +14,13 @@ return {
   },
   {
     "mrcjkb/haskell-tools.nvim",
+    ft = haskell_ft,
     dependencies = {
       "nvim-lua/plenary.nvim",
       { "nvim-telescope/telescope.nvim", optional = true },
       { "mfussenegger/nvim-dap", optional = true },
     },
-    version = "^2",
-    -- load the plugin when opening one of the following file types
-    ft = { "haskell", "lhaskell", "cabal", "cabalproject" },
+    version = "^2", -- version ^3 requires nvim >=0.9
```

Comment:
```suggestion
    version = vim.fn.has "nvim-0.8" and "^2" or "^3"
```

Reply:
beware that version 3 has a more streamlined configuration:

```
{
  'mrcjkb/haskell-tools.nvim',
  version = '^3', -- Recommended
  ft = { 'haskell', 'lhaskell', 'cabal', 'cabalproject' },
}
```

we could use a block for this as an alternative?

Reply:
I need to check init.lua locally, I may be thinking of something else.

Reply:
not sure about breaking changes, but for one: dependencies on plenary seems to be gone. I may be mistaken tho. I'm about to test your change.

Reply:
I may be doing something wrong, because if I apply the v3 line manually I still get haskell-tools 2.4.0 (working locally on my branch btw)

Reply:
something like this perhaps? at the very least, it seems to be doing something. However, haskell-tools fails to install and it complains.

```lua
    version = function()
      if vim.fn.has "nvim-0.8" then
        return "^2"
      else
        return "^3"
      end
    end,
```

<img width="1511" alt="Screen Shot 2023-11-28 at 11 48 23 AM" src="https://github.com/AstroNvim/astrocommunity/assets/3269984/5c5c0a12-ffb6-4bca-8da5-423094d1b187">

Reply:
ok I've managed. vim.fn.has requires a comparison! also: since it checks the given version and above, this makes more sense:

```
 version = vim.fn.has "nvim-0.9" == 1 and "^3" or "^2",
```

Reply:
also adding this would take care of the plenary obsolete deps:

```
      -- vim.fn.has >= nvim 0.9 removes plenary dependency
      { "nvim-lua/plenary.nvim", optional = vim.fn.has "nvim-0.9" == 1 and true or false },
```

Reply:
I'm going to force push the changes.